bundle/src/main/java/com/adobe/acs/commons/properties/filter/TemplatedFilter.java

```diff
@@ -0,0 +1,271 @@
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.Designate;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@Component(
+        service = Filter.class,
+        property = {
+                Constants.SERVICE_RANKING + ":Integer=" + Integer.MIN_VALUE,
+                EngineConstants.SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_REQUEST,
+                EngineConstants.SLING_FILTER_PATTERN + "=/content/.*"
+        },
+        configurationPolicy = ConfigurationPolicy.REQUIRE
+)
+@Designate(ocd = com.adobe.acs.commons.properties.filter.TemplatedFilter.Config.class)
+public class TemplatedFilter implements Filter {
+
+    private static final Logger log = LoggerFactory.getLogger(TemplatedFilter.class);
+
+    @Reference
+    private PropertyAggregatorService propertyAggregatorService;
+
+    private Map<String, Object> properties;
+    private List<Pattern> includePatterns;
+    private List<Pattern> excludePatterns;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        // do nothing
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        RequestPathInfo currentPathInfo = ((SlingHttpServletRequest) servletRequest).getRequestPathInfo();
+        if (StringUtils.equals(currentPathInfo.getExtension(), "json") && shouldProcess(((SlingHttpServletRequest) servletRequest).getPathInfo())) {
+            CapturingResponseWrapper capturingResponseWrapper = new CapturingResponseWrapper((SlingHttpServletResponse) servletResponse);
+            filterChain.doFilter(servletRequest, capturingResponseWrapper);
+
+            SlingHttpServletRequest slingHttpServletRequest = (SlingHttpServletRequest) servletRequest;
+
+            String currentResponse = capturingResponseWrapper.getCaptureAsString();
+            String toReturn = currentResponse;
+            try {
+                properties = propertyAggregatorService.getProperties(slingHttpServletRequest.getResource());
+                if (properties.size() > 0) {
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    JsonNode currentTree = objectMapper.readTree(currentResponse);
+                    replaceInElements(currentTree);
```

Comment:
Yeah I did this because unfortunately if there are nested items in the JSON it won't replace it properly. This also has the added benefit of resolving encoding issues if the values replaced have things like `"` in them - the `toString()` in `JsonNode` handles that stuff. I could likely find a way to make string replacement if that's desired - currently I look for `{{` and `}}` and check if the closing section is after the opening - I could probably just iterate over all the current properties and look for exact matches of `{{page_properties.jcr:title}}`, `{{page_properties.pageTitle}}`, etc

Reply:
I did some performance testing of the filter and even with 50k+ of JSON and 50+ replacements it took on average 2ms to run through this. I do think long-term it makes sense to do string replacement though, i'll get that into this PR

Reply:
given the performance you're seeing, it feels sufficient to stay as-is then